test/page.spec.js

```diff
@@ -264,6 +264,46 @@ module.exports.addTests = function({testRunner, expect, puppeteer, DeviceDescrip
     });
   });
 
+  describe('Page.createNewIsolatedContext', function() {
+    it('should work', async({page, server}) => {
+      await page.evaluate(function() {
+        window.constantVariable = new Proxy({}, {
```

Comment:
I think this test can be much simpler. Instead of messing around with proxies, just something like:
```js
await page.evaluate(() => window.dontFindThis = true);
expect(await isolatedContext.evaluate(() => window.dontFindThis)).toBe(undefined);
```